tools/check_language.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+
+# Copyright 2020 The Amber Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import fnmatch
+import os
+import re
+import sys
+
+REGEXES = [
+    r"(?i)(black|white|gr[ea]y)[-_]?list",
+    r"(?i)(first class citizen)",
+    r"(?i)(black|white|gr[ea]y)[-_]?hat",
+    r"(?i)(master|slave)",
+    r"(?i)\b(him|his|she|her|hers|man|woman)\b",
+    r"(?i)\s(he|he'd|he's|he'll)\s",
```

Comment:
People might use non-ASCII quotes (Unicode U+2019 - maybe others? Not up on this stuff.)

Reply:
Also, I'm pretty sure \s(patterns)\s won't match 'he' at start or end of line/file

Reply:
For all of the 'he' patterns you are including ^ instead of a \s which is causing them to try to match to the beginning of the line instead of leading whitespace.

Reply:
For each of those patterns both the \s and ^ version exist, so we check both.

Reply:
My bad, I was mentally parsing \she as \sshe I think.

Reply:
What version of python are you using? \u apparently was added to regex in 3.3.

Reply:
The regex works, I can put the \u2019 into an actual text file and the script picks it up. The test....doesn't. So, maybe my test data isn't utf-8?

Reply:
That is weird, strings in python 3 are unicode UTF-8 by default. Also if you have a unicode vs byte mismatch between the regex and string I believe that should throw an exception.

Possibly have the test print out the snippets and strings to see what it thinks it is working with?

Reply:
I figured it out. I was running the script with `python3` but running the tests with `python` which picks up python2. I've enabled the tests now. The script works with `python3` and I don't think it works with `python2`.